gallery/gallery-styles.css

```diff
@@ -0,0 +1,84 @@
+body {
+  background: black;
+}
+main {
+  padding-top: 125px;
+}
+
+.row {
+  display: flex;
+  flex-wrap: wrap;
+  padding: 0 4px;
+}
+
+/* Create four equal columns that sits next to each other */
+.column {
+  flex: 22%;
+  max-width: 25%;
+  padding: 0 4px;
+}
+
+.column img {
+  margin-top: 8px;
+  vertical-align: middle;
+  width: 100%;
+}
+
+.container {
+  position: relative;
+  width: 100%;
+}
+
+.image {
+  display: block;
+  width: 100%;
+  height: auto;
+}
+
+.overlay {
+  position: absolute;
+  top: 0;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  height: 100%;
+  width: 100%;
+  opacity: 0;
+  transition: 0.5s ease;
+  background-color: #000;
+}
+
+.container:hover .overlay {
+  opacity: 0.7;
+}
+
+.text {
+  color: white;
+  font-size: 20px;
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  -webkit-transform: translate(-50%, -50%);
+  -ms-transform: translate(-50%, -50%);
+  transform: translate(-50%, -50%);
+  text-align: center;
+}
```

Comment:
I see you're centering the text via the `.text` class which works on larger containers but breaks on smaller containers. if you used flexbox on `.text` parent container ,`.overlay`, you'd be able to center no matter the container size. 
on `.overlay`, try:
`box-sizing: border-box; 
display: flex;
    align-items: center;
    justify-content: center;
    padding: 0 20px;`
on `.text`, remove most of it except for
`color: white;
  font-size: 20px;
text-align: center;`